app/services/declarations/create.rb

```diff
@@ -0,0 +1,97 @@
+module Declarations
+  class Create
+    attr_reader :author,
+                :lead_provider,
+                :teacher,
+                :training_period,
+                :declaration_date,
+                :declaration_type,
+                :evidence_type,
+                :payment_statement,
+                :mentorship_period,
+                :delivery_partner
+
+    def initialize(
+      author:,
+      lead_provider:,
+      teacher:,
+      training_period:,
+      declaration_date:,
+      declaration_type:,
+      evidence_type:,
+      payment_statement:,
+      mentorship_period:,
+      delivery_partner:
+    )
+      @author = author
+      @lead_provider = lead_provider
+      @teacher = teacher
+      @training_period = training_period
+      @declaration_date = declaration_date
+      @declaration_type = declaration_type
+      @evidence_type = evidence_type
+      @payment_statement = payment_statement
+      @mentorship_period = mentorship_period
+      @delivery_partner = delivery_partner
+    end
+
+    def create
+      ActiveRecord::Base.transaction do
+        declaration = create_declaration
+
+        update_uplifts!(declaration)
+        set_eligibility!(declaration)
+        set_payment_statement!(declaration)
+        check_mentor_completion!(declaration)
+        record_declaration_created_event!(declaration)
+
+        declaration
+      end
+    end
```

Comment:
Do we need to write to the database in each step here, or could we assign attributes and save at the end?

It probably doesn't matter too much, but it felt a little unnecessary (to me) to create a declaration then update it twice in quick succession.

What do you think?